lgc/patch/PatchInOutImportExport.cpp

```diff
@@ -122,8 +124,11 @@ bool PatchInOutImportExport::runOnModule(Module &module) {
       m_shaderStage = static_cast<ShaderStage>(shaderStage);
       processShader();
 
-      // Now process the call and return instructions.
-      visitCallInsts(candidateCallees);
+      // We process input first, because we cache lots of arguments to output during visit for later processing.
+      // It will be a disaster if we visit output intrinsics first, and the cached value for output was invalidated
+      // after we process input intrinsics(consider a value read from input was exported to output).
+      visitCallInsts(inputCallees);
+      visitCallInsts(outputCallees);
```

Comment:
See above comments, we finally have
visitCallInsts(inputCallees)
visitCallInsts(outputCallees)
visitCallInsts(otherCallees)